ext/pcntl/pcntl.c

```diff
@@ -125,7 +125,15 @@ typedef psetid_t cpu_set_t;
 #include <pthread/qos.h>
 #endif
 
-#ifdef HAVE_PIDFD_OPEN
+#if defined(__linux__) && defined(HAVE_DECL_SYS_WAITID) && HAVE_DECL_SYS_WAITID == 1 && defined(HAVE_SYSCALL) && HAVE_SYSCALL == 1
```

Comment:
```suggestion
#if defined(__linux__) && defined(HAVE_DECL_SYS_WAITID) && HAVE_DECL_SYS_WAITID == 1 && defined(HAVE_SYSCALL)
```

Reply:
What about the `HAVE_SYSCALL == 1` check, this got lost in the rebasing a bit?

Reply:
@petk Check your previous message in the thread, you suggested that we move `syscall` to `AC_CHECK_FUNCS`, and then to remove `HAVE_SYSCALL == 1` as, `AC_CHECK_FUNCS` wil not define it to 0.

I don't mind either way in particular, because it's correct either way.

Reply:
That's great. Thanks. There was an actual issue here. If the macro would be defined to an empty value by the future build, the check wouldn't pass. It realistically wouldn't happen but this is a sync with other similar guard macros used in the php-src. There ideally shouldn't be `== 1` in checks with such undefined/defined macros. Like some sort of CS.

Reply:
It's a pedantic point but you're right. The [documentation](https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Generic-Functions.html) for`AC_CHECK_FUNCS`  states

> define HAVE_function (in all capitals) if it is available

and there is seemingly nothing in this contract that says that the defined value will be 1.